### PR TITLE
Feature/realmfile init

### PR DESCRIPTION
**Pull request by dereklwood**
- [x] Creates a basic realmfile for authentication if certain attributes are set.
- [x] tests for security_init recipe

### CODE REVIEW

File: recipes/security_init.rb [hunk 1]
```diff
@@ -39,3 +39,26 @@
   not_if { File.exist?(path) }
   only_if { ssl_enabled }
 end
+
+# Manage Authentication realmfile
+if node['cdap']['security']['manage_realmfile'].to_s == 'true'
+  if node.key?('cdap') && node['cdap'].key?('cdap_site') && node['cdap']['cdap_site'].key?('security.authentication.handlerClassName') &&
+     node['cdap']['cdap_site']['security.authentication.handlerClassName'] == 'co.cask.cdap.security.server.BasicAuthenticationHandler'
+    if node.key?('cdap') && node['cdap'].key?('cdap_site') && node['cdap']['cdap_site'].key?('security.authentication.basic.realmfile')
```
Comment (wolf31o2) [inline recipes/security_init.rb]: Why not continue previous if?

File: recipes/security_init.rb [hunk 2]
```diff
@@ -39,3 +39,26 @@
...
+
+      # Ensure parent directory exists
+      directory realmdir do
+        action :create
+        recursive true
+      end
+
+      # Create the realmfile
+      file realmfile do
+        content "#{node['cdap']['security']['realmfile']['username']}: #{node['cdap']['security']['realmfile']['password']}"
```
Comment (wolf31o2) [inline recipes/security_init.rb]: This only allows a single user. It's fine for testing, but should support multiple users.

Comment (dereklwood) [inline recipes/security_init.rb]: i hope nobody is using it for more than testing, but will add a attribute-driven template instead

**Comment by wolf31o2** [comment]
:+1: LGTM

**Pull request merged**